src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1910,11 +1932,16 @@ protected function preparePathInfo()
             $requestUri = '/'.$requestUri;
         }
 
-        if (null === ($baseUrl = $this->getBaseUrl())) {
+        if (null === ($baseUrl = $this->getBaseUrlReal())) {
             return $requestUri;
         }
 
-        $pathInfo = substr($requestUri, \strlen($baseUrl));
+        if (!empty($baseUrl) && 0 === strpos($requestUri, $baseUrl)) {
```

Comment:
why are you changing that code ? Is this related to your change ?

Reply:
good point. this was originally needed fix for length before I realized we need a separate `getBaseUrlReal` function. now it's not needed any more. I reverted it back to the original state.